### DISCUSSION TITLE
issue: Input not updating when using field array

### DISCUSSION THREAD

**Discussion by grzegorzkapcia** [react-hook-form/react-hook-form#10330] [Q&A]
### Version Number

7.43.5

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-get-started-forked-3t7iyn

### Steps to reproduce

1. Click "Add" to append new element to array
2. Click "Edit" button to edit added element
3. Modify value - put any string e.g. "Test 2"
4. Click "Edit Test" button
5. Value in input will be updated and set to actual value ("Test")
6. Click "Edit Test 2" button

### Expected behaviour

Input should hold value set in setp 3 "Test 2"

Actual:
Input is not refreshed and it's set to the value of previously edited element. (value in form state is correct)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment]
@grzegorzkapcia 

Your implementation with `useFieldArray` hook is not correct therefore it caused many issues on field array state and leads to wrong result.

Please always use `fields` object of `useFieldArray` hook to render field array and also CSS should be considered for hidden input element to avoid mount/unmount field array item.

Working demo: https://codesandbox.io/s/react-hook-form-get-started-forked-n5kbj0?file=/src/index.js
